dockerfiles/che/build.sh

```diff
@@ -17,15 +19,41 @@ if [ ! -d "${DIR}/../../assembly/assembly-main/target" ]; then
 fi
 
 # Use of folder
-BUILD_ASSEMBLY_ZIP=$(echo "${DIR}"/../../assembly/assembly-main/target/eclipse-che-*.tar.gz)
-LOCAL_ASSEMBLY_ZIP="${DIR}"/eclipse-che.tar.gz
+BUILD_ASSEMBLY_DIR=$(echo "${DIR}"/../../assembly/assembly-main/target/eclipse-che-*/eclipse-che-*/)
+LOCAL_ASSEMBLY_DIR="${DIR}"/eclipse-che
+
+if [ -d "${LOCAL_ASSEMBLY_DIR}" ]; then
+  rm -r "${LOCAL_ASSEMBLY_DIR}"
+fi
+
+echo "Copying assembly ${BUILD_ASSEMBLY_DIR} --> ${LOCAL_ASSEMBLY_DIR}"
+cp -r "${BUILD_ASSEMBLY_DIR}" "${LOCAL_ASSEMBLY_DIR}"
```

Comment:
why do we need to copy files instead of using the files directly

Reply:
I'm kind of -1 as it takes a lot of time to copy

Reply:
the reason why I did this is because `ln` does not work with folders so I can't just create a link as it was anymore

Reply:
btw you approved this before https://github.com/eclipse/che/pull/5880

Reply:
AFAIK `ln` is working with directories.

$ mkdir foo
$ touch foo/dummy
$ ln -s foo bar
$ ls -la bar/
-rw-r--r--   1 benoitf  wheel    0 27 sep 11:05 dummy
```

Reply:
yes, but symbolic link does not work with dockerfile `ADD` or `COPY` instructions. That was the issue. And you cant make not symbolic link to a folder

Reply:
```
$ more Dockerfile 
FROM centos
ADD bar /tmp/bar
```

$ docker build -t fb/test1 .
Sending build context to Docker daemon  3.584kB
Step 1/2 : FROM centos
 ---> 328edcd84f1b
Step 2/2 : ADD bar /tmp/bar
 ---> 5e99318d01fa
Successfully built 5e99318d01fa
Successfully tagged fb/test1:latest

$ docker run -ti -exec fb/test1 ls -la /tmp/bar/
total 8
drwxr-xr-x 2 root root 4096 Sep 27 09:42 .
drwxrwxrwt 1 root root 4096 Sep 27 09:42 ..
-rw-r--r-- 1 root root    0 Sep 27 09:10 dummy
```

Reply:
ok, when you said "symbolic link" you said "symbolic link to an external folder"

Reply:
$ mkdir -p src/2
$ touch src/test && touch src/2/test2
$ ln -s src dst

Dockerfile
```
FROM alpine
ADD dst /tmp/dst
```
build `docker build -t fb/test1 .`

```

Step 1/2 : FROM alpine
latest: Pulling from library/alpine
88286f41530e: Pull complete
Digest: sha256:f006ecbb824d87947d0b51ab8488634bf69fe4094959d935c0c103f4820a417d
Status: Downloaded newer image for alpine:latest
 ---> 76da55c8019d
Step 2/2 : ADD dst /tmp/dst
ADD failed: stat /var/lib/docker/tmp/docker-builder623285034/Users/roman/development/codenvy_projects/test/src: no such file or directory
```

Reply:
external folder, yes, sorry for confusion. So that is why `cp` was used

Reply:
@benoitf @riuvshin Are you agree with this solution? Or you have some alternative?

Reply:
an alternative would be to keep symbolic link on the file but adding a symlink inside the image
I wanted to check if running the build command twice was using the cache or not